projects/plugins/jetpack/extensions/shared/components/product-management-controls/inspector-control.js

```diff
@@ -83,6 +84,15 @@ export default function ProductManagementInspectorControl() {
 		);
 	};
 
+	useEffect( () => {
+		// If the user has manually selected a title then that should be left as-is, don't overwrite it
+		if ( isCustomTitle ) {
```

Comment:
Maybe I could get rid of this state by checking if the current `title` from the state matches anything that `getTitleByProps` could return?

Reply:
Yup, that could work, but it's fine to leave this as-is. The other benefit of getting rid of this state is that if the user undoes their changes and goes back to a default title, we'd be again changing the title according to the type/interval. In other words:

1. User changes title to "My Subscription"
2. User changes interval to "Yearly"
3. User changes title to "Yearly Subscription"
4. User changes type to "Donation"

The `isCustomTitle` check would keep the title as "Yearly Subscription", but if replace it with a check against the default values, the title will change to "Yearly Donation".

Anyway, this is an edge case, so feel free to ignore this!